contribute/guidelines_for_iron_developers.rst

```diff
@@ -0,0 +1,100 @@
+========================================
+Guidelines for OpenCMISS-Iron Developers
```

Comment:
Nothing in here is actually specific to Iron, and it repeats substantial portions of content that is elsewhere. Merge all new content into the respective sections above and remove.